### PR TITLE
#2772 issue fixed

### LINKED ISSUE #2772: ExportFormats: Use FileExtension enum
Export Formats should use the FileExtensions Enum (analog to import), too

### PR DESCRIPTION
**Pull request by PJozeph**
Fixes #2772 

----

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Screenshots added (for bigger UI changes)
- [ ] Manually tested changed features in running JabRef
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)
- [ ] If you changed the localization: Did you run `gradle localizationUpdate`?

### CODE REVIEW

**Comment by PJozeph** [comment]
ExportFormats: Use FileExtension enum #2772

**Comment by PJozeph** [comment]
okay, sorry i started travis  again, i will fix it and will come back

**Comment by Siedlerchr** [comment]
Good start, you should also take a look at the export dialog where the export formats are used to see if it can be changed there, too

**Comment by PJozeph** [comment]
Changeed the string to FileExtensions at all places.

**Comment by PJozeph** [comment]
hmmm... again most of the test failed, how can I checked where i broke the code ?

**Comment by PJozeph** [comment]
How can i check in Travis where i broke the  code ?

**Comment by Siedlerchr** [comment]
@PJozeph  Select one of the travis jobsm e.g. modernizer and click on it it out:
Seems like you forgot some pieces of code. Eclipse should have already warned you about build errors,
But the error log on travis looks odd. I restarted the job
https://travis-ci.org/JabRef/jabref/jobs/319061181

File: src/main/java/org/jabref/logic/exporter/BibTeXMLExportFormat.java [hunk 1]
```diff
@@ -20,22 +20,7 @@
 import javax.xml.datatype.XMLGregorianCalendar;
 import javax.xml.namespace.QName;
 
-import org.jabref.logic.importer.fileformat.bibtexml.Article;
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/logic/exporter/BibTeXMLExportFormat.java]: The problem are these removed imports (and similarly further down)

Comment (PJozeph) [reply src/main/java/org/jabref/logic/exporter/BibTeXMLExportFormat.java]: i dont have org.jabref.logic.importer.fileformat.bibtexml folder

**Comment by LinusDietz** [comment]
There is still a problem with the imports. try running `./gradlew check` on your local computer to avoid the delay through travis

**Comment by Siedlerchr** [comment]
In Eclipse you can run CTRL+Shift+O to organize your imports

**Comment by PJozeph** [comment]
that's right,but  since the project  does not have org.jabref.logic.importer.fileformat.bibtexml folder the  CTRL+Shift+O will remove those imports @Siedlerchr

**Comment by Siedlerchr** [comment]
Then you have messed up sth. I will take a look at your branch

**Comment by Siedlerchr** [comment]
Ah okay, you have an import name clash. If you simply remove the first import of java.io.File it should work again. (I tested it locally)

**Comment by PJozeph** [comment]
removed the first import of java.io.File but still got the same error. Furthermore  i still dont understand why there is no org.jabref.logic.importer.fileformat.bibtexml folder @Siedlerchr 

jabref

**Comment by PJozeph** [comment]
I dont understand this. The ModsExportFormat and BibTeXMLExportFormat with full of compiler error there are many Classes that cannot be resolved. I still dont have org.jabref.logic.importer.fileformat.bibtexml package.

Removed the first java.io.File in the ModsExportFormat and BibTeXMLExportFormat class but got the same issue. 

What i see after running .**_/gradlew check_**  is that the build fails due to this compiler errors.  


jabref2
t the same issue. 
jabref

**Comment by PJozeph** [comment]
jabrefgradlewcheck

**Comment by Siedlerchr** [comment]
I created a patch file which fixes the imports and your build:
[0004-fix-imports.patch.txt](https://github.com/JabRef/jabref/files/1576371/0004-fix-imports.patch.txt)

To import it, remove the .txt extension and do:
`git am < 0004-fix-imports.patch`

**Comment by PJozeph** [comment]
unfortunately , no it does not 


jabrefpatch

**Comment by tobiasdiez** [comment]
This really looks like a problem with your setup. The code in question (bibtexml/...) is automatically generated if I'm not mistaken. Can you try to run `./gradlew cleanEclipse`, `./gradlew eclipse` and ``./gradlew generateSource`?

**Comment by PJozeph** [comment]
i worked, but it failed again, and no idea why

**Comment by PJozeph** [comment]
i decided to rebuild the working environment. But the build failed  and i got a  exception as :   Execution failed for task ':junitPlatformTest'.
> Process 'command '/usr/lib/jvm/java-8-oracle/bin/java'' finished with non-zero exit value 1

**Pull request closed without merging**